### PR TITLE
For #7467: Update ktlint to 0.47.0.

### LINKED ISSUE #7467: Update code quality tools
This should include updating ktlint and detekt, and also fixing issues and removing entries from baselines (if any).

### PR DESCRIPTION
**Pull request by mcarare**
Replace deprecated disabled_rules with ktlint_disabled_rules.
Fix existing issues.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### GitHub Automation
Fixes #7467

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.kt [hunk 1]
```diff
@@ -101,7 +101,8 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         window.statusBarColor = ContextCompat.getColor(this, android.R.color.transparent)
         when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
             Configuration.UI_MODE_NIGHT_UNDEFINED, // We assume light here per Android doc's recommendation
-            Configuration.UI_MODE_NIGHT_NO, -> {
+            Configuration.UI_MODE_NIGHT_NO,
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:104]: No strong opinions here, but I think this trailing comma is not useful when matching?

File: app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchOverlay.kt [hunk 2]
```diff
@@ -40,7 +40,8 @@ fun SearchOverlay(
 
     when (state.value) {
         is State.Disabled,
-        is State.NoSuggestionsAPI, -> {
+        is State.NoSuggestionsAPI,
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchOverlay.kt:43]: Same question here.

**Review by jonalmeida — APPROVED** [review]
LGTM, just a small question added, but feel free to land this.

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:104]: Here is just the same argument as any other trailing commas, you can add another matching value for that case without changing more than one row, preserving blame.

**Pull request merged**